internal/beater/java_attacher/java_attacher.go

```diff
@@ -145,7 +149,13 @@ func (j *JavaAttacher) Run(ctx context.Context) error {
 		return fmt.Errorf("java attacher is disabled")
 	}
 
-	// Non-Windows: run discovery and attachment until context is closed.
+	tempDir, err := j.createAttacherTempDir()
+	if err != nil {
+		j.logger.Errorf("failed to create a Java attacher temp dir: %v. Using the bundled attacher jar instead", err)
+	} else {
+		defer j.deleteTempDir(tempDir)
+	}
```

Comment:
I didn't bother looking into all possible errors returned by everything I used to create the temp dir, however, if we default to using the bundled attacher, it will still work if the application user is the same as the Elastic Agent user. I agree this is not likely in production, but could be a useful fallback for users that just want to explore on some test env or even locally on their laptop.

Reply:
> but could be a useful fallback for users that just want to explore on some test env or even locally on their laptop.

OK. This is not a good enough reason to increase the code complexity, so please just return the error. Failure to create a temporary file suggests a more serious issue, like the disk being full.

Reply:
OK, will do.
Explaining the rationale- the way I see it, ad-hoc trial is a very important use case and an error here can be anything, for example such raised by the user's laptop OS implementation indicating it does not support it.

Reply:
Still needs changing, will change that shortly

Reply:
Currently, we still fallback to the bundled attacher jar.
Pro: we may be able to attach to a specific user but not to other
Con: we'll try every time we see a JVM for that user (but only once per JVM)

LMK if you want something else, for example:
- bail and exit the attacher upon the first failure
- cache failure per user using a special value (so that we can still attach to other users' JVMs)

Reply:
Please log & cache the failure, and continue attempting for other users.